server/src/main/java/org/opensearch/action/admin/indices/alias/TransportIndicesAliasesAction.java

```diff
@@ -131,45 +136,97 @@ protected void clusterManagerOperation(
-        List<AliasAction> finalActions = new ArrayList<>();
+        List<AliasAction> finalActions = resolvedAliasActions(request, state, true);
+        if (finalActions.isEmpty() && false == actions.isEmpty()) {
+            throw new AliasesNotFoundException(
+                actions.stream().flatMap(a -> Arrays.stream(a.getOriginalAliases())).collect(Collectors.toSet()).toArray(new String[0])
+            );
+        }
+        request.aliasActions().clear();
+        IndicesAliasesClusterStateUpdateRequest updateRequest = new IndicesAliasesClusterStateUpdateRequest(unmodifiableList(finalActions))
+            .ackTimeout(request.timeout())
+            .clusterManagerNodeTimeout(request.clusterManagerNodeTimeout());
+
+        indexAliasesService.indicesAliases(updateRequest, new ActionListener<ClusterStateUpdateResponse>() {
+            @Override
+            public void onResponse(ClusterStateUpdateResponse response) {
+                listener.onResponse(new AcknowledgedResponse(response.isAcknowledged()));
+            }
+
+            @Override
+            public void onFailure(Exception t) {
+                logger.debug("failed to perform aliases", t);
+                listener.onFailure(t);
+            }
+        });
+    }
+
+    @Override
+    public ResolvedIndices resolveIndices(IndicesAliasesRequest request) {
+        try {
+            Set<String> indices = new HashSet<>();
+
+            for (AliasAction aliasAction : resolvedAliasActions(request, clusterService.state(), false)) {
+                if (aliasAction instanceof AliasAction.Add addAliasAction) {
+                    indices.add(addAliasAction.getIndex());
+                    indices.add(addAliasAction.getAlias());
+                } else if (aliasAction instanceof AliasAction.Remove removeAliasAction) {
+                    indices.add(removeAliasAction.getIndex());
+                    indices.add(removeAliasAction.getAlias());
+                } else if (aliasAction instanceof AliasAction.RemoveIndex removeIndexAction) {
+                    // TODO special action
+                    indices.add(removeIndexAction.getIndex());
+                }
+            }
+
+            return ResolvedIndices.of(indices);
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            // This should not happen if validate=false is passed to resolvedAliasActions()
+            throw new RuntimeException(e);
+        }
+    }
+
+    private List<AliasAction> resolvedAliasActions(IndicesAliasesRequest request, ClusterState state, boolean validate) throws Exception {
+        List<AliasAction> result = new ArrayList<>();
         // Resolve all the AliasActions into AliasAction instances and gather all the aliases
-        Set<String> aliases = new HashSet<>();
-        for (IndicesAliasesRequest.AliasActions action : actions) {
+        for (IndicesAliasesRequest.AliasActions action : request.aliasActions()) {
```

Comment:
Why is this change necessary exactly? 

I noticed that we were instantiating the ActionRequestMetadata in the apply method of ActionFilter:

```
this.action.filters[i].apply(task, actionName, request, new ActionRequestMetadata<>(action, request), listener, this);
```

On the surface, it seems to be redundant to pass both the transport action and request to the constructor there. Would it be possible to have an addition method on the transport action? i.e. `action.getRequestMetadata`?